Fix flaky meeting spec partially

cbliard opened the pull request:
failing spec: `modules/meeting/spec/features/structured_meetings/structured_meeting_update_flash_spec.rb`

When run locally, it does not wait enough after saving an agenda item and it sometimes creates synchronizations issues such as:

- click on the menu icon of an agenda item too early, then the items are updated by the turbo response and the menu is closed, so it fails to find the ".Overlay" element. It is retried so it's not that big of a problem, but it's not ideal as it waits for nothing.

- save an agenda item on a browser window, then switch to another tab and expect the update notification to be displayed. Well sometimes it's not saved yet, so the notification is not displayed, and the test fails.

- A variant of the previous one: save an agenda item on a browser window, then switch to another tab and reload the page. Sometimes the save is not finished yet, so the page reloads but the agenda item is not there yet.

The fix is to wait for the save to complete. Either by waiting for the record to be created or updated, and then for the element to be displayed. Or also by just sleeping 1 second because I could not find a better synchronization mechanism.

Note that this still not fixes the flakiness where the `travel_to` seems to not be effective anymore when the recurring meeting is created. See run https://github.com/opf/openproject/actions/runs/17466279904/job/49602727897 for instance.

cbliard committed:
Fix flaky meeting spec partially

failing spec: `modules/meeting/spec/features/structured_meetings/structured_meeting_update_flash_spec.rb`

When run locally, it does not wait enough after saving an agenda item
and it sometimes creates synchronizations issues such as:

- click on the menu icon of an agenda item too early, then the items are
  updated by the turbo response and the menu is closed, so it fails to
  find the ".Overlay" element. It is retried so it's not that big of a
  problem, but it's not ideal as it waits for nothing.

- save an agenda item on a browser window, then switch to another tab
  and expect the update notification to be displayed. Well sometimes
  it's not saved yet, so the notification is not displayed, and the test
  fails.

- A variant of the previous one: save an agenda item on a browser
  window, then switch to another tab and reload the page. Sometimes the
  save is not finished yet, so the page reloads but the agenda item is
  not there yet.

The fix is to wait for the save to complete. Either by waiting for the
record to be created or updated, and then for the element to be
displayed. Or also by just sleeping 1 second because I could not find a
better synchronization mechanism.

Note that this still not fixes the flakiness where the `travel_to` seems
to not be effective anymore when the recurring meeting is created. See run
https://github.com/opf/openproject/actions/runs/17466279904/job/49602727897
for instance.

mrmir commented on modules/meeting/spec/support/pages/meetings/show.rb:
```diff
@@ -370,13 +375,36 @@ def add_agenda_item_to_backlog(type: MeetingAgendaItem, &)
+      MeetingAgendaItem.maximum(:id)
+    end
+
+    # warning: does not work when using `travel_to` because time is stopped.
```
Is checking `lock_version` here not possible? That should be incremented every update right?

cbliard replied:
Ah true, that's a better idea. I'll update the code.

mrmir reviewed (approved):
Thanks for the changes Christophe!

cbliard committed:
Automatically wait for agenda item update in specs

`Pages::Meetings::Show#edit_agenda_item` now saves by default, except if
`save: false` is specified.

Waiting for a change in `locked_version` is better than waiting for a
change in `updated_at`.

cbliard commented:
Spec `./spec/features/users/create_spec.rb:139` is passing locally.
Merging.
